hs_core/views/__init__.py

```diff
@@ -712,45 +712,23 @@ def delete_multiple_files(request, shortkey, *args, **kwargs):
 
 def delete_resource(request, shortkey, *args, **kwargs):
     res, _, user = authorize(request, shortkey, needed_permission=ACTION_TO_AUTHORIZE.DELETE_RESOURCE)
-
-    res_title = res.metadata.title
-    res_id = shortkey
-    res_type = res.resource_type
-    resource_related_collections = [col for col in res.collections.all()]
-    owners_list = [owner for owner in res.raccess.owners.all()]
-    ajax_response_data = {'status': 'success'}
-    try:
-        hydroshare.delete_resource(shortkey)
-    except ValidationError as ex:
-        if request.is_ajax():
-            ajax_response_data['status'] = 'error'
-            ajax_response_data['message'] = str(ex)
-            return JsonResponse(ajax_response_data)
-        else:
-            request.session['validation_error'] = str(ex)
-            return HttpResponseRedirect(request.META['HTTP_REFERER'])
-
-    # if the deleted resource is part of any collection resource, then for each of those collection
-    # create a CollectionDeletedResource object which can then be used to list collection deleted
-    # resources on collection resource landing page
-    for collection_res in resource_related_collections:
-        o=CollectionDeletedResource.objects.create(
-             resource_title=res_title,
-             deleted_by=user,
-             resource_id=res_id,
-             resource_type=res_type,
-             collection=collection_res
-             )
-        o.resource_owners.add(*owners_list)
-
-    post_delete_resource.send(sender=type(res), request=request, user=user,
-                              resource_shortkey=shortkey, resource=res,
-                              resource_title=res_title, resource_type=res_type, **kwargs)
-
     if request.is_ajax():
-        return JsonResponse(ajax_response_data)
+        task_id = get_resource_delete_task(shortkey)
+        if not task_id:
+            pre_delete_resource.send(sender=type(res), request=request, user=user, resource_shortkey=shortkey,
+                                     resource=res, resource_type=res.resource_type, **kwargs)
+            task = delete_resource_task.apply_async((shortkey, user.username))
+            task_id = task.task_id
+        task_dict = get_or_create_task_notification(task_id, name='resource delete', payload=shortkey,
+                                                    username=user.username)
+        return JsonResponse(task_dict)
     else:
-        return HttpResponseRedirect('/my-resources/')
+        try:
+            hydroshare.delete_resource(shortkey)
```

Comment:
why is this handled differently than an ajax request?

Reply:
@sblack-usu This is for backward compatibility with delete resource REST API call but now that I think about it, REST API delete resource must have its own view function, i.e., not go through this view function. If that is indeed the case, I can remove this conditional check and handling. Let me know what you think.